### PR TITLE
UI: Add new theme, update theme capabilities

### PR DESCRIPTION
**Pull request by Fenrirthviti**
I have always felt that the out of the box themes for OBS were quite
lacking, and have spent a lot of time going through and sorting out the
difficulties with the current setup. I've added a new themeID parameter to
several elements that were otherwise impossible to target with QSS in a
theme. Since Qt has pushed for the use of QML over QSS at this point,
these should be considered workarounds. Included is the theme I was
working on that can serve as a base. I'm hoping to encourage others to
make their own themes, so we can grow the available themes for OBS.

I am happy for any feedback on the theme itself, or other updates that
can me made to make creating new themes easier overall.

### CODE REVIEW

File: UI/data/themes/Rachni.qss [hunk 1]
```diff
@@ -0,0 +1,1032 @@
+/***************************************************************************/
+/*   Copyright (C) 2017-2018 by Joel Bethke <joel.bethke@gmail.com>        */
+/*                                                                         */
+/*                                                                         */
+/*   This program is free software: you can redistribute it and/or modify  */
+/*   it under the terms of the GNU General Public License as published by  */
+/*   the Free Software Foundation, either version 2 of the License, or     */
+/*   (at your option) any later version.                                   */
```
Comment (RytoEX) [inline UI/data/themes/Rachni.qss]: License:  Is this actually okay?  I know that OBS itself is strictly GPLv2, so derivative works must have a compatible license, and this (in the GPLv2 option) is still compatible...  @dodgepong ?

File: UI/data/themes/Rachni.qss [hunk 2]
```diff
@@ -0,0 +1,1032 @@
...
+	border-image: url(./Rachni/down_arrow_disabled.png);
+	height: 10px;
+	width: 10px;
+	subcontrol-position: bottom;
+	subcontrol-origin: margin;
+}
+
+QScrollBar::sub-line:vertical:hover,
+QScrollBar::sub-line:vertical:on {
+
```
Comment (RytoEX) [inline UI/data/themes/Rachni.qss]: Remove this blank line.

File: UI/data/themes/Rachni.qss [hunk 3]
```diff
@@ -0,0 +1,1032 @@
...
+/********************************/
+
+QPushButton#streamButton,
+QPushButton#recordButton,
+QPushButton[themeID="replayBufferButton"],
+QPushButton#modeSwitch,
+QPushButton#settingsButton,
+QPushButton#exitButton {
+	background-color: rgb(0, 188, 212); /* Cyan (Primary) */
+
```
Comment (RytoEX) [inline UI/data/themes/Rachni.qss]: Remove this blank line.

File: UI/data/themes/Rachni.qss [hunk 4]
```diff
@@ -0,0 +1,1032 @@
...
+QPushButton:hover#moveEffectFilterUp {
+	background-color: rgba(240, 98, 146, 0.5); /* Pink (Secondary) */
+	margin: 0;
+	padding: 0;
+	border-radius: 2px;
+	border: none;
+	outline: none;
+}
+
+/**************************/
```
Comment (RytoEX) [inline UI/data/themes/Rachni.qss]: This box here seems a bit mangled.  Give it some help to be a real box?

File: UI/data/themes/Rachni.qss [hunk 5]
```diff
@@ -0,0 +1,1032 @@
...
+	color: rgb(255, 148, 194); /* Light Pink (Secondary Light) */
+	font-weight: bold;
+}
+
+QLabel#errorLabel {
+	color: rgb(186, 45, 101); /* Dark Pink (Secondary Dark) */
+	font-weight: bold;
+}
+
+/********************/
```
Comment (RytoEX) [inline UI/data/themes/Rachni.qss]: This box here seems a bit mangled. Give it some help to be a real box?

File: UI/data/themes/Rachni.qss [hunk 6]
```diff
@@ -0,0 +1,1032 @@
...
+
+QSplitter::handle:horizontal {
+	width: 1px;
+}
+
+QSplitter::handle:vertical {
+	height: 1px;
+}
+
+/*******************/
```
Comment (RytoEX) [inline UI/data/themes/Rachni.qss]: This box here seems a bit mangled. Give it some help to be a real box?

File: UI/data/themes/Rachni.qss [hunk 7]
```diff
@@ -0,0 +1,1032 @@
...
+
+QAbstractItemView {
+	background-color: rgb(35, 38, 41); /* Dark Gray */
+}
+
+QToolTip {
+	border: 1px solid rgb(118, 121, 124); /* Light Gray */
+	background-color: rgb(49, 54, 59); /* Blue-gray */
+	color: rgb(240, 98, 146); /* Pink (Secondary) */
+}
```
Comment (RytoEX) [inline UI/data/themes/Rachni.qss]: This file should end with a newline.

File: UI/data/themes/Rachni.qss (hunk 1 above)
Comment (dodgepong) [inline UI/data/themes/Rachni.qss]: That's correct. If this is GPLv2, then it's compatible with OBS's GPLv2 license. This is fine, as far as I can tell.

Comment (dodgepong) [inline UI/data/themes/Rachni.qss]: Compare to the license at the top of this random libobs file I grabbed: https://github.com/jp9000/obs-studio/blob/master/libobs/obs-audio.c#L1-L16

Comment (RytoEX) [inline UI/data/themes/Rachni.qss]: Ah, you're right.  I was thinking of [this commit](https://github.com/jp9000/obs-studio/commit/8298fa4dc7108f2e4323f26f04ab19fc6d43e376), but forgot it was GPLv2+ and not strictly GPLv2.  My mistake.

**Pull request closed without merging**